### ISSUE TITLE
REGEXP_LIKE has counterintuitive behavior when matching on multiline strings

### ISSUE THREAD

**Issue by alec-heif** [trinodb/trino#8827]
I recently got bitten by a bug where `REGEXP_LIKE` seems to not support a pattern that covers multiple lines.

Consider the following inline table which contains a single `VARCHAR` cell with multiple lines:
```
SELECT * 
FROM (VALUES ('foo
bar')) 
AS t (query);
```
This will produce the following output when run in the Trino CLI:
```
 query
-------
 foo
 bar
(1 row)
```

Now let's filter `query` by a regex.

Filtering to `.*foo.*` works:

```
SELECT * FROM (
  SELECT * 
  FROM (VALUES ('foo
bar')) 
  AS t (query))
WHERE REGEXP_LIKE(query, '.*foo.*');

 query
-------
 foo
 bar
(1 row)
```

And also filtering to `.*bar.*` works too!

```
SELECT * FROM (
  SELECT * 
  FROM (VALUES ('foo
bar')) 
  AS t (query))
WHERE REGEXP_LIKE(query, '.*bar.*');

 query
-------
 foo
 bar
(1 row)
```

But bizarrely filtering to `.*foo.*bar.*` does not?!

```
SELECT * FROM (
  SELECT * 
  FROM (VALUES ('foo
bar')) 
  AS t (query))
WHERE REGEXP_LIKE(query, '.*foo.*bar.*');

 query
-------
(0 rows)
```

Either of the following seem like ok solutions to me:

1) Update the `REGEXP_LIKE` documentation to clarify this caveat (maybe something as simple as "regexes matches across multiple lines are not supported")
2) Fix the problem.

Obviously 2 will result in slightly less user confusion but it's also a fairly niche use case so maybe not worth putting the trouble to fix.

**Comment by findinpath** [comment]
I agree that the `REGEXP_LIKE` function does definitely need some improvement in the documentation.

Note that the `REGEXP_LIKE` implementation is based on the library

https://github.com/airlift/joni

which is a port of the library 

https://github.com/kkos/oniguruma/


Have a look over the tests for this function to get a feeling how the implementation works:

https://github.com/trinodb/trino/blob/master/core/trino-main/src/test/java/io/trino/operator/scalar/AbstractTestRegexpFunctions.java


In the meanwhile, for your specific problem, this could be a solution:

```
SELECT * FROM (
   SELECT * 
   FROM (VALUES ('foo
 bar')) 
   AS t (query))
WHERE REGEXP_LIKE(query, '.*foo\n.*bar.*');


query 
-------
 foo   
 bar   
(1 row)

```
